.github/workflows/main.yml

```diff
@@ -0,0 +1,137 @@
+
+      - name: test with coverage (ubuntu)
+        id: test_Ubuntu
+        if: matrix.os == 'ubuntu-latest'
+        run: xvfb-run --auto-servernum npm run test:coverage
+
+      - name: test (macos, windows)
+        if: matrix.os != 'ubuntu-latest'
+        run: npm test
+
+      - name: ui test (ubuntu)
+        id: uiTest_Ubuntu
+        if: matrix.os == 'ubuntu-latest'
+        run: xvfb-run --auto-servernum npm run ui-test
+
+      - name: ui test (macos, windows)
+        id: uiTest_MacOS_Windows
+        if: matrix.os != 'ubuntu-latest'
+        run: npm run ui-test
+
+      - name: vsce-package
+        run: vsce package
+
+      - name: Store Camel Language Server log
+        uses: actions/upload-artifact@v3
+        # TODO: Exists? In the same directory on windows/macos?
+        if: failure()
+        with:
+          name: ${{ matrix.os }}-${{ matrix.version }}-camel-lsp-log
+          path: /tmp/log-camel-lsp.log
+
+      - name: Store VS Code Logs (Ubuntu)
+        uses: actions/upload-artifact@v3
+        if: failure() && matrix.os == 'ubuntu-latest'
+        with:
+          name: ${{ matrix.os }}-${{ matrix.version }}-vscode-logs
+          path: ~/.config/Code/logs/*
+
+      - name: Store VS Code Logs (Macos)
+        uses: actions/upload-artifact@v3
+        if: failure() && matrix.os == 'macos-latest'
+        with:
+          name: ${{ matrix.os }}-${{ matrix.version }}-vscode-logs
+          path: ~/Library/Application Support/Code/logs/*
+
+      - name: Store VS Code Logs (Windows)
+        uses: actions/upload-artifact@v3
+        if: failure() && matrix.os == 'windows-latest'
+        with:
+          name: ${{ matrix.os }}-${{ matrix.version }}-vscode-logs
+          path: $env:USERPROFILE\AppData\Code\logs\*
+
+      - name: Store UI test log
+        uses: actions/upload-artifact@v3
+        if: failure() && (steps.uiTest_Ubuntu.outcome == 'failure' || steps.uiTest_MacOS_Windows.outcome == 'failure')
+        with:
+          name: ${{ matrix.os }}-${{ matrix.version }}-ui-test-logs
+          path: uitest-resources/settings/logs/*
+
+      - name: Store UI Test Screenshots
+        uses: actions/upload-artifact@v3
+        if: failure() && (steps.uiTest_Ubuntu.outcome == 'failure' || steps.uiTest_MacOS_Windows.outcome == 'failure')
+        with:
+          name: ${{ matrix.os }}-${{ matrix.version }}-ui-test-screenshots
+          path: uitest-resources/screenshots/*.png
+
+      - name: Store Test Coverage
+        uses: actions/upload-artifact@v3
+        if: (steps.test_Ubuntu.outcome == 'success')
+        with:
+          name: ${{ matrix.os }}-${{ matrix.version }}-test-coverage
+          path: coverage/*
+
+      - name: SonarCloud Scan
+        uses: sonarsource/sonarcloud-github-action@master
+        if: matrix.os == 'ubuntu-latest' && github.event_name == 'push'
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          SONAR_TOKEN: ${{ secrets.SONAR_TOKEN }}
```

Comment:
Do you have enough rights to configure the SONAR_TOKEN?

Reply:
Hi @apupier, @mlorinc, @djelinek 

No, I do not, but there is the more fundamental problem between GH actions and sonarcloud I want to discuss. 

I faced the [issue](https://community.sonarsource.com/t/sonar-cannot-be-run-on-pr-from-a-fork/69229) with GH actions:
```
Note: With the exception of GITHUB_TOKEN, secrets are not passed to the runner when a workflow is triggered from a forked repository.
```
So it is not possible to run that step on PR from a fork

I have set up it to run on push events  (a merged PR will always create a push event to the branch)

So here is the main question: **how do we want to see sonarcloud in vscode projects with GH actions**?

**1. [DAP style](https://github.com/camel-tooling/camel-dap-client-vscode):** 

Pros:

-  Auto-check on PRs with the bot notifications after each commit

Cons:

-   [js and ts do not have test-coverage auto scan](https://docs.sonarcloud.io/enriching/test-coverage/javascript-typescript-test-coverage/) => no results about test coverage => 
- no default `Quality Gates` result

**2. DAP style for PRs with the extra Push steps on the main (actually in that PR):** 

Pros:

-  Auto-check on PRs with the bot notifications after each commit (need to set sonarcloud bot in repo settings, I do not have access)
- Test coverage result (in PRs in logs and artifacts and on main Push in sonarcloud)
- Default `Quality Gates` value

Cons:

- Extra steps,  extra changes

**3. DAP style with customized `Quality Gates` without test-coverage:** 

Pros:

- As in the first scenario 
- Customized `Quality Gates` value

Cons:

- No test-coverage results

**4. DAP style for PRs with extra pipeline and test-coverage ([issue-workaround](https://community.sonarsource.com/t/how-to-use-sonarcloud-with-a-forked-repository-on-github/7363/29)):** 

Pros:

- As in the second scenario

Cons:

- Extra pipeline, extra changes

What do you think is the better option, or maybe you see some extra possibilities?

Reply:
as a first iteration, I think we can keep it simple and use the simpler config as DAP style without coverage.
Other scenarii can be discussed in a separate issue after that.